app/views/products/charts.html.erb

```diff
@@ -2,27 +2,41 @@
 <head>
   <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
   <script type="text/javascript">
-    google.charts.load('current', {'packages':['corechart']});
-    google.charts.setOnLoadCallback(drawChart);
+  google.charts.load('current', {'packages':['corechart']});
+  google.charts.setOnLoadCallback(drawChart);
+  function drawChart() {
 
-    function drawChart() {
-      var data = google.visualization.arrayToDataTable(
-          <%= product_prices('pen').to_json.html_safe %>
-      );
+var datasArray = []
 
-      var options = {
-        title: 'Product Prices',
-        curveType: 'function',
-        legend: { position: 'bottom' }
-      };
+    <%@product.map do|product| %>
+    var data = google.visualization.arrayToDataTable(
+      <%= product_prices(product.name).to_json.html_safe %>
+    );
+    datasArray.push(data);
+    <%end%>
 
-      var chart = new google.visualization.LineChart(document.getElementById('curve_chart'));
+    var options = {
+      title: 'Product Prices',
+      curveType: 'function',
+      legend: { position: 'bottom' }
+    };
 
-      chart.draw(data, options);
+
+
+
+    for(var f = 0; f < datasArray.length; f++){
+      var randomId = Math.random(10);
+      var addedElement = document.createElement('div');
+      addedElement.id = randomId;
+      document.body.appendChild(addedElement);
+      var chart = new google.visualization.LineChart(document.getElementById(randomId));
```

Comment:
you do not need to getElementById as you have it already in the variable `addedElement`